packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -646,6 +720,140 @@ class OpenUpwardsPageTransitionsBuilder extends PageTransitionsBuilder {
   }
 }
 
+/// Used by [PageTransitionsTheme] to define a horizontal [MaterialPageRoute] page
+/// transition animation that looks like the default page transition
+/// used on Android U.
+///
+/// {@tool dartpad}
+/// This example shows the default page transition on Android.
+///
+/// ** See code in examples/api/lib/material/page_transitions_theme/page_transitions_theme.3.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [FadeUpwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android O.
+///  * [OpenUpwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Andoird P.
+///  * [ZoomPageTransitionsBuilder], which defines the default page transition
+///    that's similar to the one provided in Android Q.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches native iOS page transitions.
+///  * [PredictiveBackPageTransitionsBuilder], which defines a page
+///    transition that allows peeking behind the current route on Android.
+///  * [FadeForwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android U.
+class FadeForwardsPageTransitionsBuilder extends PageTransitionsBuilder {
+  /// Constructs a page transition animation that matches the transition used on
+  /// Android U.
+  const FadeForwardsPageTransitionsBuilder({
+    this.backgroundColor,
+  });
+
+  /// The background color during transition between two routes.
+  ///
+  /// When a new page fades in and the old page fades out, this background color
+  /// helps avoid a black background between two page.
+  ///
+  /// Defaults to [ColorScheme.surface]
+  final Color? backgroundColor;
```

Comment:
Do we have use case where people want to set a different color other than surface? it seems whichever color they pick will cause a color flash

Reply:
I think so! If the page has its own color, changing the `backgroundColor` to the same color can avoid the color flash. I noticed this problem in page_transition_theme.0.dart, the two pages has their own `Scaffold.backgroundColor`, the first page has blueGrey as its background, so if we want to use the new builder and no color flash, we should also set the background color to `Colors.blueGrey`.